### PR TITLE
refactor: push Connect schema down

### PR DESCRIPTION
**Pull request by big-andy-coates**
### Description 

Following on from the fixes and refactors done in #6188, this commit pushes down the use of the Connect schema type to lower levels of the code. Higher levels of the code now deal with `LogicalSchema`, `PersistentSchema` or just a `List<SimpleColumn>`.

This moves us closer to removing the Connect schema from the code base, except in the Serde code that deals with connect formats.

`LogicalSchema` and `PersistentSchema` no longer know about the Connect schema type. Calls to retrieve the Connect schema from these types have been replaced with a util function that can convert a list of columns into a Struct Connect schema.  As more code moves away from the Connect schema these util function calls will slowly be removed.

A good sign this is an improvement is that this PR deletes more code than it adds, and now error messages use SQL types, not connect types.

### Testing done 

usual

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Review by agavra — APPROVED** [review]
LGTM! there's a perf consideration that I flagged for going forward, but it's only related to your change in that it might be harder to fix after your change than before

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/api/impl/KeyValueExtractor.java [hunk 1]
```diff
@@ -35,9 +37,13 @@
...
 
-  public static Struct extractKey(final JsonObject values, final LogicalSchema logicalSchema,
-      final SqlValueCoercer sqlValueCoercer) {
-    final Struct key = new Struct(logicalSchema.keyConnectSchema());
+  public static Struct extractKey(
+      final JsonObject values,
+      final LogicalSchema logicalSchema,
+      final SqlValueCoercer sqlValueCoercer
+  ) {
+    final ConnectSchema keySchema = ConnectSchemas.columnsToConnectSchema(logicalSchema.key());
```
[the remark this answers is not in the document]
Comment (big-andy-coates) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/api/impl/KeyValueExtractor.java:45]: FYI: This is currently only used for inserts.

Totally agree we should be thinking about perf here. However, as the plan is to move away from using Connect's `Struct` type for keys, this conversion is only temporary.

Regardless, I'll pull the conversion higher up so its cached.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutor.java [hunk 2]
```diff
@@ -1048,10 +1050,12 @@ private static KsqlException invalidWhereClauseException(
   }
 
   private static Struct asKeyStruct(final Object keyValue, final PhysicalSchema physicalSchema) {
-    final Field keyField = Iterables
-        .getOnlyElement(physicalSchema.keySchema().connectSchema().fields());
+    final ConnectSchema keySchema = ConnectSchemas
+        .columnsToConnectSchema(physicalSchema.keySchema().columns());
```
[the remark this answers is not in the document]
Comment (big-andy-coates) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutor.java:1054]: As above, this conversion is only temporary.  In this case, it's a per-request cost, so can't be amalgamated.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/api/impl/KeyValueExtractor.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (big-andy-coates) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/api/impl/KeyValueExtractor.java:45]: See: https://github.com/confluentinc/ksql/pull/6206/files#r489274132

**Pull request merged**